### PR TITLE
Docker support for Hygieia

### PR DESCRIPTION
**Pull request by tyrion70**
A simple setup to run Hygieia in a Docker container

### CODE REVIEW

**Comment by srid99** [comment]
@tyrion70 for UI setup, instead of using 'gulp serve' can you change it to use apache server ?

**Comment by srid99** [comment]
Gave a try by running `createdocker` script. The build went well but it failed when it tried `docker rm hygieia`.

```
Step 39 : CMD /usr/bin/supervisord -n
 ---> Running in 742586701a53
 ---> e970f0d0a90b
Removing intermediate container 742586701a53
Successfully built e970f0d0a90b
Error response from daemon: no such id: hygieia
FATA[0000] Error: failed to remove one or more containers 
```

**Comment by tyrion70** [comment]
Hmm that kinda makes sense. The first time u run it there is nog container yet. I'll fix that tonight thanks

**Comment by tyrion70** [comment]
Updated the createdocker script so it won't give an error the first time (it did start the container eventually right @srid99).
Also added Apache support, please anyone take a good look at what I did, I'm not sure if I have the document root completely correct. Its currently set to the /UI/dist folder now. I tried the src folder, but then it needed a symlink to include bower_components and it was missing css.
A Reverse proxy for /api was added to the apache configuration as well.

**Comment by srid99** [comment]
@tyrion70 Correct. It had error but the build was success and the container was running.

For the Apache support, what you did is correct. `dist` folder and the reverse proxy should do the job.

**Comment by tabladrum** [comment]
@tyrion70: can you submit the CLA? Thanks

**Comment by tyrion70** [comment]
already did yesterday :) u didn't get it?

File: Docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#
+# Caveats: the mongo datafiles are a local volume 
+#   (/root/rocker/mongodb in the example) that needs to be 
+#   changed accordingly. The first time you start Hygieia 
+#   the db.dashboarddb database won't exist. 
+#   This might cause failures.
+
+FROM ubuntu:14.04
+
+MAINTAINER Peter van Mourik <tyrion70@gmail.com>
```
Comment (jdamick) [inline Docker/Dockerfile]: should change this over to the group: Hygieia@capitalone.com

**Comment by jdamick** [comment]
So in looking at this vs. pull #115 - to be honest i'd like to see something in the middle.. This pull has jenkins,sonar, etc all in the same container and lacks configurability.  The other pull has them separated out but lacks the collectors.. So I'm thinking about merging the best of both of them into a branch and then once it's solidified a bit merge it back to master.  Sound fair?

**Comment by tyrion70** [comment]
Hey! 

It's actually its different containers. One container is for the complete Hygieia solution including mongodb and apache. It also allows you to start other containers for sonar and jenkins (and mysql) using compose. 

Nevertheless there are some nice things in the other PR as well so feel free to solidify and merge it together :)

What do you mean by lack of configurability by the way? All config files needed are in the docker folder and can be modified before building the container. 

Let me know if you need any help merging or have any questions about what I did..

**Comment by pjaol** [comment]
I've started down the path of dockerizing some of the collectors, will add those as separate pulls

**Comment by jdamick** [comment]
closed by pull #129

**Pull request closed without merging**